From: tomasz-karczewski-red <tomasz.karczewski@redembedded.com>
Date: Tue, 15 Mar 2022 15:35:13 +0100
Subject: [PATCH] ONEMPERS-367 Avoid sending 'unknown method' response

When websocket header is read & processed but no payload
has yet been received on the socket, it 'parses' 0 bytes
as empty json object & responds with 'unknow method' response.
This fix avoids that scenario.

---
 Source/websocket/WebSocketLink.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Source/websocket/WebSocketLink.h b/Source/websocket/WebSocketLink.h
index 6f4c3ab2..937aa54f 100644
--- a/Source/websocket/WebSocketLink.h
+++ b/Source/websocket/WebSocketLink.h
@@ -644,7 +644,9 @@ POP_WARNING()
                                 result += static_cast<uint16_t>(headerSize + payloadSizeInControlFrame); // actualDataSize
 
                             } else {
-                                _parent.ReceiveData(&(dataFrame[result + headerSize]), actualDataSize);
+                                if (actualDataSize > 0) {
+                                   _parent.ReceiveData(&(dataFrame[result + headerSize]), actualDataSize);
+                                }
 
                                 result += (headerSize + actualDataSize);
                             }
-- 
2.25.1

